app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt

```diff
@@ -0,0 +1,155 @@
+import com.github.se.eventradar.R
+import com.github.se.eventradar.ui.BottomNavigationMenu
+import com.github.se.eventradar.ui.component.*
+import com.github.se.eventradar.ui.navigation.NavigationActions
+import com.github.se.eventradar.ui.navigation.TOP_LEVEL_DESTINATIONS
+import com.github.se.eventradar.ui.navigation.getTopLevelDestination
+
+@Composable
+fun HostingScreen(navigationActions: NavigationActions) {
+  val mockEvents = getMockEvents()
+  var viewMapOrListIndex by remember { mutableIntStateOf(0) }
+  ConstraintLayout(modifier = Modifier.fillMaxSize().testTag("hostingScreen")) {
+    val (logo, title, divider, eventList, bottomNav, buttons) = createRefs()
+    Logo(
+        modifier =
+            Modifier.fillMaxWidth()
+                .constrainAs(logo) {
+                  top.linkTo(parent.top, margin = 32.dp)
+                  start.linkTo(parent.start, margin = 16.dp)
+                }
+                .testTag("logo"))
+    Title(
+        modifier =
+            Modifier.testTag("myHostedEventsTitle")
+                .constrainAs(
+                    title,
+                    {
+                      top.linkTo(logo.bottom, margin = 16.dp)
+                      start.linkTo(parent.start)
+                      end.linkTo(parent.end)
+                    }))
+    HorizontalDivider(
+        modifier = Modifier.constrainAs(divider, { top.linkTo(title.bottom, margin = 10.dp) }))
+    if (viewMapOrListIndex == 0) {
+      EventList(
+          mockEvents,
+          Modifier.fillMaxWidth().constrainAs(eventList) {
+            top.linkTo(divider.bottom, margin = 8.dp)
+            start.linkTo(parent.start)
+            end.linkTo(parent.end)
+          })
+    } else {
+      // To Do (Map View)
+    }
+
+    Row(
+        modifier =
+            Modifier.constrainAs(ref = buttons) {
+                  bottom.linkTo(bottomNav.top)
+                  centerHorizontallyTo(parent)
+                }
+                .testTag("floatingActionButtons")
+                .fillMaxWidth()
+                .padding(16.dp),
+        horizontalArrangement = Arrangement.Absolute.Left,
+        verticalAlignment = Alignment.CenterVertically) {
+          CreateEventFab(onClick = { /*TODO*/}, modifier = Modifier.testTag("createEventFab"))
+          Spacer(modifier = Modifier.width(16.dp))
+          var viewToggleIcon =
+              if (viewMapOrListIndex == 0) Icons.Default.Place else Icons.AutoMirrored.Filled.List
+          ViewToggleFab(
+              modifier = Modifier.testTag("viewToggleFab"),
+              onClick = {
+                viewMapOrListIndex = if (viewMapOrListIndex == 0) 1 else 0
+                viewToggleIcon =
+                    if (viewMapOrListIndex == 0) Icons.Default.Place
+                    else Icons.AutoMirrored.Filled.List
+              },
+              iconVector = viewToggleIcon)
+        }
+    BottomNavigationMenu(
+        onTabSelected = { tab -> navigationActions.navigateTo(tab) },
```

Comment:
nit: you can use `navigationActions::navigateTo` to avoid the lambda expression.